src/types/ownsMany.js

```diff
@@ -46,34 +46,92 @@ class OwnsMany extends MultiModel {
         return model;
     }
 
-    requestMany (options: ?Object) {
+    // @override
+    requestMany (options: Object = {}): any {
         const model = this._instantiateModel();
 
         model
             .getAdapter()
             .findAll(this.Model, this._parent, options, (response) => {
-                this.apply(response, options);
+                return this
+                    .apply(response)
+                    .applyQueryResults(response, options);
             });
+
+        return this;
     }
 
-    apply (data : Array<Object>) {
+    apply (data : Array<Object>): Object {
         data.forEach((item) => {
             const model = this._instantiateModel();
 
             this._models[item.id] = model;
             model.apply(item);
         });
+
+        return this;
+    }
+
+    applyQueryResults (data : Array<Object>, options: Object = {}): Object {
+        const idKey = this.Model.getFieldId();
+
+        const ids = data.map((modelData) => {
+            return modelData[idKey];
+        });
+
+        this.requestCache.set(options, ids);
+
+        return this;
+    }
+
+    all (): Array<Object> {
+        return this.map.map((id) => {
+            return this._models[id];
+        });
+    }
+
+    add (model: Object, index: ?number): Object {
+        super.add(...arguments);
+
+        this._models[model.id] = model;
+
+        return this;
     }
 
     toJSON (options : Object = {}): any {
         if (options.recursive) {
-            return Object.keys(this._models).map((id) => {
-                return this._models[id].toJSON(options);
+            return this.all().map((model) => {
+                return model.toJSON(options);
+            });
+        }
+
+        return [];
+    }
+
+    query (options: Object = {}): Array<any> {
```

Comment:
At least in our case, we'd never have a case where we query a hasMany, but I could see this maybe being valuable. It would probably need a slightly different implementation for hasMany, though